libs/feature/search/src/lib/utils/service/aggregations.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core'
+import { SearchApiService } from '@geonetwork-ui/data-access/gn4'
+import { ElasticsearchService } from '@geonetwork-ui/util/shared'
+import { Observable } from 'rxjs'
+import { filter, map } from 'rxjs/operators'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AggregationsService {
+  constructor(
+    private esService: ElasticsearchService,
+    private searchApiService: SearchApiService
+  ) {}
+
+  getAggregation(fieldName: string): Observable<any> {
```

Comment:
I don't think the name of this method is meaning full. 
There are tons of way to fetch ES aggregations and your payload is hardcoded here.

Your method should be `getFullSearchWithoutTemplatesTermAggregationsOrderedByKey` :)
I don't know what should be the strategy here, you should be able to pass the ordering key at least.

Note that aggregations can be something else than `terms`

Reply:
Hum, good point that helped my understanding:-) I tried to find a compromise.